### DISCUSSION TITLE
Upgrade from PrimeNG 15 to v16 (and 17) breaks some styles of p-inputNumber (Ultima Theme)

### DISCUSSION THREAD

**Discussion by TheTiGuR** [primefaces/community#553] [PrimeNG Templates]
After going through the upgrade, the p-inputNumber with up/down controls is broken, style-wise:
![image](https://github.com/primefaces/community/assets/542067/45565974-17ea-4df3-9640-c6bd1ce74540)
It previously looked like this: 
![image](https://github.com/primefaces/community/assets/542067/7507e57f-20ee-445b-a98d-dbb2dd3b4674)
The buttons for up and down continue to work fine, but obviously are incorrectly sized.

**Comment by cagataycivici** [comment]
Could be due to CSS layers, do you have custom CSS that styles buttons e..g normalize.css, bootstrap, tailwind?

**Reply by TheTiGuR** [reply]
I might have some individual styles that might be causing problems, but overall everything is just primeng + primeflex + Ultima. What section on the styling of  the component can I show you to help diagnose?

**Reply by ThoSap** [reply]
@cagataycivici yes, this is due to the `@layer primeng` implementation introduced in PrimeNG `16.4.0`.
https://github.com/primefaces/primeng-sass-theme/issues/47

But this has absolutely **nothing to do with custom styles**, the same issue appears if I just update from PrimeNG `16.3.1` to **>= 16.4.0** in any of the templates I bought (Ultima, Diamond, Mirage, Freya, Apollo, Atlantis) without touching anything.

**Reply by ThoSap** [reply]
@TheTiGuR also see https://github.com/orgs/primefaces/discussions/269#discussioncomment-7331010
Use PrimeNG `16.3.1` in the meantime, you won't have any template-related issues with this version.

**Reply by ThoSap** [reply]
See also:
https://github.com/primefaces/primeng/issues/13757
https://github.com/primefaces/primeng/issues/13780
https://github.com/primefaces/primeng/issues/14072
https://github.com/orgs/primefaces/discussions/130

**Reply by TheTiGuR** [reply]
Thanks for the info @ThoSap ! I will probably just deal with it for now, pending the update since it happens in only 1 location. But, hopefully the themes get updated sooner rather than later.

**Reply by cetincakiroglu** [reply]
Hi,

We're planning to update next-gen templates to the v17 soon.
